### PR TITLE
Issue 2 set up continuous integration e g travis ci

### PR DESCRIPTION
**Pull request by a-t-0**
-  Created instructions to run Truffle unit testing on solidity contracts
-  Restructured repository hierarchy to adhere to default solidity CI setup (with Truffle).
-  Added `public` keyword to several functions to remove compilation errors where `public/private` was not defined.
- Changed `abstract` contract templates to actual contracts to remove compilation errors during testing.
- Verified testing works on the example contract.

### CODE REVIEW

**Review by v-bosch — APPROVED** [review]
looks good :) not sure why you need to specify the array's length everywhere now, i didnt have problems with that. Anyways, good work!

**Comment by a-t-0** [comment]
> ure why you need to specify the array's length everywhere now, i didnt have problems with that. Anyways, good work!

Thank you, the specification was required because the test file would raise a compilation error if it was not. And if you specify it for one variable it needs to be propagated to the entire chain that that variable touches.

File: contracts/SolveNQueens.sol [hunk 1]
```diff
@@ -22,16 +22,15 @@ contract SolveNQueens {
         testContract.test(owner);
     }
 
-    function main(bool[][] memory unsolved_board) public returns (bool[][] memory solved_board) {
-        (bool a, bool[][] memory solution) = solveNQ(x, N);
+    function main(bool[4][4] memory unsolved_board) public returns (bool[4][4] memory solved_board) {
```
Comment (mdroogh) [inline contracts/SolveNQueens.sol:25]: Why is it necessary to strictly limit the size to 4?

File: truffle-config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
```
Comment (mdroogh) [inline truffle-config.js:1]: Why commit code which is fully commented out?

**Review by mdroogh — APPROVED** [review]

File: contracts/SolveNQueens.sol (hunk 1 above)
Comment (a-t-0) [reply contracts/SolveNQueens.sol:25]: The code currently throws an error if the dimensions are not specified.

File: truffle-config.js (hunk 2 above)
Comment (a-t-0) [reply truffle-config.js:1]: Excellent question, this is auto-generated code from the metacoin example truffle unbox, my preference would be for now to keep it, if needed for reference until a better understanding of the complete functionality is developed.

**Pull request merged**